src/components/directory-content.js

```diff
@@ -525,13 +595,22 @@ const DirectoryContent = () => {
                                     name: e.name,
                                 };
                             });
-
-                            setChildrenMetadata(metadata);
                         }
                     );
+                })
+                .then(() => {
```

Comment:
should be a then in the fetchContingencyListsInfos level no ? otherwise the metadata will be possibly partial